File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.cpp
#include <string>
#include <vector>
using namespace std;

int gcd(int a, int b){
    if(b == 0)
        return a;
    else 
        return gcd(b, a % b);
}

vector<int> solution(int numer1, int denom1, int numer2, int denom2) {
    vector<int> answer;
 
    int numer = numer1*denom2 + numer2*denom1; // 분자
    int denom = denom1*denom2; // 분모
    
    int div = gcd(numer, denom); // 최대공약수
    
    numer /= div;
    denom /= div;
    
    answer = {numer, denom};
    
    return answer;
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int [2];
        int max = 1; // 최대공약수
        int limit;
        
        int u = numer1 * denom2 + numer2 * denom1; // 분자
        int d = denom1 * denom2; // 분모
        
        if(u > d) limit = d;
        else limit = u;
        
        // 최대공약수를 지정
        for(int i = 1; i <= limit; i++){
            if(u % i == 0 && d % i == 0)
                max = i;
        }
        
        answer[0] = u / max;
        answer[1] = d / max;
       
        
        return answer;
    }
}